### PR TITLE
UI improvement

### PR DESCRIPTION
**Pull request by Sanlorng**
## Update all UI designs for the app. 
**The new design refers to [material design official documentation ](https://material.io) as much as possible**

### 1. **Home Page**

- Replace the drawer with the top tab, and in the future use Navigation Rail(medium) or DawerButton(large) for large screens

Home Page

### 2. **Bench Page**

- Update item design refers to [OneLineItem, TwoLineItem, ThreeLineItem](https://material.io/components/lists#specs)

Bench Page

Bench Page 2

### 3. **History Page**

- Update item design refers to [Expandable item](https://material.io/components/lists#specs). In the future, after updating to Compose 1.1, an expand/shirink animation will be added by using AnimateVisibility

History Page

### 4. **Setting Page**

- Update item design referes to [TwoLineItem](https://material.io/components/lists#specs)

Setting Page

- Update TargetPathDialog design referes to [Dialog](https://material.io/components/dialogs#specs)

Target path dialog

- Update SingleChoiceDialog deisgn referes to [Single choice dialog](https://material.io/components/dialogs#specs)

Single choice dialog

- Update TextInputDialog design referes to [Dialog](https://material.io/components/dialogs#specs)

Text input dialog

- Update AboutDialog design referes to [Dialog](https://material.io/components/dialogs#specs)
About dialog

### CODE REVIEW

File: app/src/main/java/io/github/devriesl/raptormark/ui/benchmark/TestItem.kt [hunk 1]
```diff
@@ -17,25 +18,54 @@ fun TestItem(
...
+    val minHeight = when (lineCount) {
+        1 -> 48.dp
+        2 -> 64.dp
+        else -> 88.dp
+    }
+    Box(
         modifier = Modifier
-            .padding(horizontal = 16.dp, vertical = 4.dp)
+            .padding(horizontal = 16.dp)
+            .defaultMinSize(minHeight = minHeight)
```
Comment (DevriesL) [inline app/src/main/java/io/github/devriesl/raptormark/ui/benchmark/TestItem.kt:37]: Why we need to hard code `defaultMinSize` calculated by `lineCount `?

File: app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt [hunk 2]
```diff
@@ -23,36 +29,67 @@ fun TextInputDialog(
...
+                )
+            )
+        }
 
-        Surface(
+        val focusRequester = remember { FocusRequester() }
+        val inputService = LocalTextInputService.current
+        LaunchedEffect(Unit) {
+            //Wait a little time to make sure the input service wakes up the keyboard
+            delay(300)
```
Comment (DevriesL) [inline app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt]: Can we replace `300` as a const val?

**Comment by DevriesL** [comment]
Thanks for your Pull Request with great UI improvement, just two minor comments, please check.

File: app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt (hunk 2 above)
Comment (Sanlorng) [reply app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt]: that's ok

File: app/src/main/java/io/github/devriesl/raptormark/ui/benchmark/TestItem.kt (hunk 1 above)
Comment (Sanlorng) [reply app/src/main/java/io/github/devriesl/raptormark/ui/benchmark/TestItem.kt:37]: @DevriesL refere to the [list item doc](https://material.io/components/lists#specs). One line item min height is 48dp, Two line item min height is is 64dp, Three line minheight is 88dp. The title of the test item is two lines under the portrait screen and one line in the horizontal screen, and latency is also taken into account, so we need to know the number of lines to determine the minimum height of the test item

File: app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt (hunk 2 above)
Comment (Sanlorng) [reply app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt]: @DevriesL [this commit](https://github.com/DevriesL/RaptorMark/pull/3/commits/a634bb8d0777a93b05bc2fd56655eecdc4661995) contains the change

Comment (DevriesL) [reply app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt]: I prefer to use `companion object` and `SHOW_SOFT_KEYBOARD_DELAY_TIME` 😄

Comment (Sanlorng) [reply app/src/main/java/io/github/devriesl/raptormark/ui/setting/TextInputDialog.kt]: @DevriesL [0f7d64d](https://github.com/DevriesL/RaptorMark/pull/3/commits/0f7d64d9a000f9fe31fa903040d6aaafa8c5ee54) contains the change. Compose first-party control libraries tend to use **Default** objects to name control constant objects

File: app/src/main/java/io/github/devriesl/raptormark/ui/setting/TargetPathDialog.kt [hunk 3]
```diff
@@ -21,62 +26,88 @@ fun TargetPathDialog(
...
-        Surface(
+        val selectCustomPath = remember { mutableStateOf(customValue.isNotEmpty()) }
+        val focusRequester = remember { FocusRequester() }
+        val inputService = LocalTextInputService.current
+        LaunchedEffect(key1 = selectCustomPath.value) {
+            if (selectCustomPath.value) {
+                delay(TextInputDialogDefault.SHOW_SOFT_KEYBOARD_DELAY_TIME)
+                inputService?.showSoftwareKeyboard()
+                focusRequester.requestFocus()
+            }else {
```
Comment (DevriesL) [inline app/src/main/java/io/github/devriesl/raptormark/ui/setting/TargetPathDialog.kt]: Minor formatting issue~

Comment (Sanlorng) [reply app/src/main/java/io/github/devriesl/raptormark/ui/setting/TargetPathDialog.kt]: > Minor formatting issue~

@DevriesL Please review [formatting change](https://github.com/DevriesL/RaptorMark/pull/3/commits/7b413737de8e269b83cb6734eed0e17485662602)

**Pull request merged**